### PR TITLE
Added copy constructor to ResourceLimitedVector. [6411]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Backport of #681

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/466/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/423/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/464/)

**Comment by MiguelCompany** [comment]
Liveliness tests seem to be also failing on base branch (1.8.x). See [this manual run](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual/479/)

File: include/fastrtps/utils/collections/ResourceLimitedVector.hpp [hunk 1]
```diff
@@ -95,6 +95,44 @@ class ResourceLimitedVector
...
+     * Assignment operator.
+     *
+     * Makes the contents of the ResourceLimitedVector the same as the ones in other
+     * ResourceLimitedVector.
+     *
+     * @param other ResourceLimitedVector from where contents are copied.
+     *
+     * @return a reference to this ResourceLimitedVector.
+     */
+    ResourceLimitedVector& operator = (const ResourceLimitedVector& other)
```
Comment (LuisGP) [inline include/fastrtps/utils/collections/ResourceLimitedVector.hpp]:
```suggestion
    ResourceLimitedVector& operator =(
            const ResourceLimitedVector& other)
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/488/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/442/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/485/)

**Pull request merged**